Enable using MO/A10/D10 for NeoPixels DMA on the QT PY

apendley opened the pull request:
I have had cases with the QT PY where I've wished to use IR receivers or other hardware that generally cannot be used at the same time as NeoPixels. Since the QT PY uses a SAMD21 however, it should be capable of allowing the use of DMA for NeoPixels, freeing up the CPU for the other hardware. This PR enables using DMA NeoPixels on pin M0/A10/D10, however doing so comes with the caveat that SPI peripherals cannot be used at the same time as DMA NeoPixels given their use of the MOSI pin. I think this trade-off can be worth it in many cases, as the QT PY prominently features I2C as an alternative, and SPI may not be as critical for most people. Also, there is precedent in enabling the MOSI pin at the expense of SPI peripherals for several other boards, so it seems like a trade-off that would be consistent.

ladyada commented:
im ok with it, @PaintYourDragon any thoughts?

PaintYourDragon reviewed (commented):
Seems good. I’d be OK adding SERCOM3 as well, since external flash is optional, though requires soldering to pad as there’s no edge pin for this. I’ll leave this open for the time being and can approve/merge whichever way you’d prefer, just say the word. Thanks!

apendley commented:
I really like the idea of enabling SERCOM3! I can add this to the PR if you'd like, I think it could be quite useful :)

apendley commented:
I went ahead and added SERCOM3 after testing that it works, and that MOSI still works. Also I updated the comments to (hopefully) better explain the choices. I can revert this change if you'd prefer however. Thanks!

PaintYourDragon reviewed (approved):
Fantastic. Thank you!
